modal/cli/import_refs.py

```diff
@@ -92,136 +93,149 @@ def import_file_or_module(file_or_module: str):
+        if not accept_local_entrypoints and isinstance(cli_command.runnable, LocalEntrypoint):
+            return False
+        if not accept_web_endpoints and cli_command.is_web_endpoint:
+            return False
+        return True
 
-Registered functions and local entrypoints on the selected app are:
-{registered_functions_str}
-"""
-    raise click.UsageError(help_text)
+    res = []
+    for cli_command in cli_commands:
+        if not _is_accepted_type(cli_command):
+            continue
 
+        if name_prefix in cli_command.names:
+            # exact name match
+            res.append(cli_command)
+            continue
 
-def _show_no_auto_detectable_app(app_ref: ImportRef) -> None:
-    object_path = app_ref.object_path
-    import_path = app_ref.file_or_module
-    error_console = Console(stderr=True)
-    error_console.print(f"[bold red]Could not find Modal app '{object_path}' in {import_path}.[/bold red]")
-
-    if object_path is None:
-        guidance_msg = (
-            f"Expected to find an app variable named **`{DEFAULT_APP_NAME}`** (the default app name). "
-            "If your `modal.App` is named differently, "
-            "you must specify it in the app ref argument. "
-            f"For example an App variable `app_2 = modal.App()` in `{import_path}` would "
-            f"be specified as `{import_path}::app_2`."
-        )
-        md = Markdown(guidance_msg)
-        error_console.print(md)
+        if not name_prefix:
+            # no name specified, return all reachable runnables
+            res.append(cli_command)
+            continue
+
+        # partial matches e.g. app or class name - should we even allow this?
+        prefix_matches = [x for x in cli_command.names if x.startswith(f"{name_prefix}.")]
+        if prefix_matches:
+            res.append(cli_command)
+    return res
 
 
 def import_app(app_ref: str) -> App:
     import_ref = parse_import_ref(app_ref)
 
+    # TODO: default could be to just pick up any app regardless if it's called DEFAULT_APP_NAME
+    #  as long as there is a single app in the module?
+    import_path = import_ref.file_or_module
+    object_path = import_ref.object_path or DEFAULT_APP_NAME
+
     module = import_file_or_module(import_ref.file_or_module)
-    app = get_by_object_path(module, import_ref.object_path or DEFAULT_APP_NAME)
+
+    if "." in object_path:
+        raise click.UsageError(f"{object_path} is not a Modal App")
+
+    app = getattr(module, object_path)
 
     if app is None:
-        _show_no_auto_detectable_app(import_ref)
+        error_console = Console(stderr=True)
+        error_console.print(f"[bold red]Could not find Modal app '{object_path}' in {import_path}.[/bold red]")
+
+        if object_path is None:
+            guidance_msg = Markdown(
+                f"Expected to find an app variable named **`{DEFAULT_APP_NAME}`** (the default app name). "
+                "If your `modal.App` is named differently, "
```

Comment:
Something I'm a little confused by: is this guiding users to to the _App name_ (e.g. `App("my-app")`) or to the _variable name_ that points at the App?

Reply:
Maybe could rephrase (e.g. "If your `modal.App` is assigned to a different variable name") to reduce the chance of confusion 🤷‍♂️